src/Disks/ObjectStorages/S3/S3ObjectStorage.cpp

```diff
@@ -288,7 +294,12 @@ void S3ObjectStorage::getDirectoryContents(const std::string & path,
 
     Aws::S3::Model::ListObjectsV2Request request;
     request.SetBucket(bucket);
-    request.SetPrefix(path);
+    /// NOTE: if you do "ls /foo" instead of "ls /foo/" it will return only
+    /// "/foo" itself without any underlying nodes.
```

Comment:
This comment is unclear. Probably you want `path` to be always ended with `/` because you don't want to find e.g. "/foo2"

Reply:
Any suggestions? For me it looks understandable, and quite frankly I don't see how the comment can be interpreted in the way that you wrote.

Reply:
I don't know about "should" but this is how API works.

```sh
$ find . -type f
./foo/bar/baz/bam
$ s3cmd put -r foo s3://clickhouse-s3-plain-test/
upload: 'foo/bar/baz/bam' -> 's3://clickhouse-s3-plain-test/foo/bar/baz/bam'  [1 of 1]
 0 of 0     0% in    0s     0.00 B/s  done

$ aws s3api list-objects-v2 --bucket clickhouse-s3-plain-test --prefix foo/ --delimiter /
None
COMMONPREFIXES  foo/bar/
$ aws s3api list-objects-v2 --bucket clickhouse-s3-plain-test --prefix foo --delimiter /
None
COMMONPREFIXES  foo/
```

Reply:
By saying "should" I meant from my understanding of the API which is not the same as the API itself of course. I didn't check it thoroughly. Thanks for the clarification!

Reply:
Yes, it is quite a thing, hence I put a comment.